packages/valory/skills/dynamic_nft_abci/models.py

```diff
@@ -60,13 +60,13 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         )
         leaderboard_api_key = self._ensure("leaderboard_api_key", kwargs, str)
         leaderboard_params = self._ensure("leaderboard_param_template", kwargs, str)
-        leaderboard_params.replace(
-            "<leaderboard_layers_range>", self.leaderboard_layers_range
+        leaderboard_params = (
+            leaderboard_params.replace(
+                "<leaderboard_layers_range>", self.leaderboard_layers_range
+            )
+            .replace("<leaderboard_points_range>", self.leaderboard_points_range)
+            .replace("<leaderboard_api_key>", leaderboard_api_key)
```

Comment:
Bug: I was not properly storing the modified params into the leaderboard_params var.